pyproject.toml

```diff
@@ -0,0 +1,72 @@
+[build-system]
+requires = ["setuptools"]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "histogrammar"
+description = "Composable histogram primitives for distributed data reduction"
+keywords = [
+    "pandas",
+    "spark",
+    "data-science",
+    "data-analysis",
+    "statistics",
+    "python",
+    "jupyter",
+    "ipython"
+]
+readme = "README.rst"
+requires-python = ">=3.9"
+authors = [{ name = "Jim Pivarski (DIANA-HEP)", email = "pivarski@fnal.gov" }, { name = "Max Baak", email = "maxbaak@gmail.com" }]
+maintainers = [{ name = "Max Baak", email = "maxbaak@gmail.com" }]
+license = { type = "Apache Software License v2", file = "LICENSE" }
+dependencies = [
+    "numpy",
+    "tqdm",
+    "joblib>=0.14.0"
+]
+classifiers = ["Development Status :: 5 - Production/Stable",
+    "Environment :: Console",
+    "Intended Audience :: Science/Research",
+    "License :: OSI Approved :: Apache Software License",
+    "Topic :: Scientific/Engineering :: Information Analysis",
+    "Topic :: Scientific/Engineering :: Mathematics",
+    "Topic :: Scientific/Engineering :: Physics",
+]
+dynamic = ["version"]
+
+[project.optional-dependencies]
+test = [
```

Comment:
Can you add the optional dependencies:
`pandas = [
    "pandas"
]

spark = [
  "pyspark>=3.1; python_version <= '3.11'",
]
`

Reply:
Done